src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/VariableDeclarations.cs

```diff
@@ -83,17 +83,51 @@ public static VariableDeclarations GenerateDeclarationsForUnmanagedToManaged(Bou
             foreach (BoundGenerator marshaller in marshallers.NativeParameterMarshallers)
             {
                 TypePositionInfo info = marshaller.TypeInfo;
-                if (info.IsNativeReturnPosition || info.IsManagedReturnPosition)
+                if (info.IsNativeReturnPosition)
+                    continue;
+
+                if (info.IsManagedReturnPosition)
                     continue;
 
                 // Declare variables for parameters
                 AppendVariableDeclarations(variables, marshaller, context, initializeToDefault: initializeDeclarations);
+
+                {
+                    // We need to use the 'out' value - This should be removed once the ownership behavior is fixed
+                    var boundaryBehavior = marshaller.Generator.GetValueBoundaryBehavior(info, context);
+                    if (marshaller.Generator.UsesNativeIdentifier(marshaller.TypeInfo, context)
+                        && boundaryBehavior is not
+                            (ValueBoundaryBehavior.NativeIdentifier or ValueBoundaryBehavior.CastNativeIdentifier))
+                    {
+                        if (MarshallerHelpers.IsMidlOutBehavior(marshaller.TypeInfo, context))
+                        {
+                            string outlocal = context.GetAdditionalIdentifier(info, "out");
+                            initializations.Add(MarshallerHelpers.CreateDiscardStatement(outlocal));
+                        }
+                    }
+                }
             }
 
             if (!marshallers.IsManagedVoidReturn)
             {
                 // Declare variables for stub return value
                 AppendVariableDeclarations(variables, marshallers.ManagedReturnMarshaller, context, initializeToDefault: initializeDeclarations);
+                {
+                    var marshaller = marshallers.ManagedReturnMarshaller;
+                    var info = marshaller.TypeInfo;
```

Comment:
Local versus using `marshaller.TypeInfo`, it is confusing why or when it should be done.